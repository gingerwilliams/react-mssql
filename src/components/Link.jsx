import React from "react";

const STATUS = {
    HOVERED: "hovered",
    NORMAL: "normal"
};

class Link extends React.Component {
    constructor(props) {
        super(props);
    
        this.onMouseEnter = this.onMouseEnter.bind(this);
        this.onMouseLeave = this.onMouseLeave.bind(this);
    
        this.state = {
          class: STATUS.NORMAL,
        };
      }
    
      onMouseEnter() {
        this.setState({class: STATUS.HOVERED});
      }
    
      onMouseLeave() {
        this.setState({class: STATUS.NORMAL});
      }
    
      render() {
        return (
          <a
            className={this.state.class}
            href={this.props.page || '#'}
            onMouseEnter={this.onMouseEnter}
            onMouseLeave={this.onMouseLeave}
          >
            {this.props.children}
          </a>
        );
      }
}

export default Link;
