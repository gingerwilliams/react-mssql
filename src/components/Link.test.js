import React from 'react';
import renderer from 'react-test-renderer';
import Link from './Link';

test('title', () => { 
    const component = renderer.create(
        <Link page="http://www.facebook.com">Facebook</Link>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseEnter();

    // // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();

    // // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});




