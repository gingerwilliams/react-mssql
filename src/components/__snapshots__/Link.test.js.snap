// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`title 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`title 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
